src/integration-test/java/io/aiven/kafka/connect/http/IntegrationTest.java

```diff
@@ -297,17 +276,11 @@ void testFailingEvery3rdRequest() throws ExecutionException, InterruptedExceptio
             sendFuture.get();
         }
 
-        TestUtils.waitForCondition(
-            () -> bodyRecorderHandler.recorderBodies().size() >= expectedBodies.size(),
-            10_000,
-            "All requests received by HTTP server"
-        );
-        log.info("Recorded request bodies: {}", bodyRecorderHandler.recorderBodies());
-        assertIterableEquals(expectedBodies, bodyRecorderHandler.recorderBodies());
+        await("All expected requests received by HTTP server")
+                .atMost(Duration.ofSeconds(15)).pollInterval(Duration.ofMillis(100))
+                .until(() -> bodyRecorderHandler.recorderBodies().size() >= expectedBodies.size());
 
-        log.info("{} HTTP requests were expected, {} were successfully delivered",
-            expectedBodies.size(),
-            bodyRecorderHandler.recorderBodies().size());
+        assertThat(bodyRecorderHandler.recorderBodies()).containsExactlyInAnyOrderElementsOf(expectedBodies);
```

Comment:
FYI, I stopped after the second assert, should probably check all the asserts

Reply:
Fixed